apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -924,14 +924,27 @@ public partial class BigtableClientImpl
     {
         private const string ResourcePrefixHeader = "google-cloud-resource-prefix";
 
+        private CallSettings _idempotentMutateRowSettings;
+
+        partial void OnConstruction(Bigtable.BigtableClient grpcClient, BigtableSettings effectiveSettings, ClientHelper clientHelper)
+        {
+            _idempotentMutateRowSettings = effectiveSettings.IdempotentMutateRowSettings;
+        }
+
         partial void Modify_ReadRowsRequest(ref ReadRowsRequest request, ref CallSettings settings) =>
             ApplyResourcePrefixHeader(ref settings, request.TableName);
 
         partial void Modify_SampleRowKeysRequest(ref SampleRowKeysRequest request, ref CallSettings settings) =>
             ApplyResourcePrefixHeader(ref settings, request.TableName);
 
-        partial void Modify_MutateRowRequest(ref MutateRowRequest request, ref CallSettings settings) =>
+        partial void Modify_MutateRowRequest(ref MutateRowRequest request, ref CallSettings settings)
+        {
+            if (settings == null && request.IsIdempotent())
```

Comment:
I don't think we can use settings in this way - we want to merge the call timing in with settings, if the user hasn't specified their own call timing, I think.
(The call settings can contain a cancellation token or auth for example, which shouldn't affect the retry semantics.)

Basically, the settings here aren't quite the same as the ones in ApiCall, which is what the defaults are used for.

You could potentially use

```csharp
if (request.IsIdempotent()
{
    settings = _idempotentMutateRowSettings.MergedWith(settings);
}
```

@chrisdunelm Does that sound right to you?

Reply:
Ok, thanks. I'll use this (unless Chris disagrees).